### ISSUE TITLE
Calendar fails to display tasks with due dates with a red error message

### ISSUE THREAD

**Issue by TheLinuxEXP** [elementary/calendar#734]
### What Happened?

I removed my CalDav online account and re-added it.
When I opened Calendar again, it displayed an error message:

Unable to connect to “TLE Tasks": Backend factory for source *b458ba0756188af7f450b3924884eff423ecedff” and extension “Calendar” cannot be found.

![Screenshot from 2021-12-20 14-14-47](https://user-images.githubusercontent.com/54111876/146772785-57a8cc66-4b77-40ba-b77e-b7585bf16681.png)

If I uncheck the tasks lists in the Calendars list in Calendar, the error disappears.

Note: if the tasks lists are checked (supposedly visible), the Date and time indicators crashes the whole top panel when opened.

### Steps to Reproduce

1. REmoved my CalDav online account (nextcloud)
2. Added it again with URL, Login and app password
3. Login works and list of calendars appears, including the tasks lists
4. Open Calendar
5. The banner appears

### Expected Behavior

There shouldn't be an error message, and the tasks with due dates should display in the calendar, as well as in the date and time indicator.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Hardware Info

Mesa drivers (integrated AMD graphics from a ryzen 4800H).
Nothing special in the setup
THe caldav calendars and tasks lists are provided through my own Nextcloud server, using https
The "Tasks" app manages to open the tasks lists and interact with the without any issue.

**Comment by mcclurgm** [comment]
We don’t currently support tasks in calendar, so I’m not sure why you would be seeing task lists in calendar. Do they appear at all in the tasks app?

We have an existing issue for adding tasks support, so for that part see #748.

**Comment by TheLinuxEXP** [comment]
Yes, tasks do appear in the Tasks app as well, I'd honestly prefer them not to be in the Calendar, and for a while they didn't, I don't really know why they started appearing there!

**Comment by mcclurgm** [comment]
Ok, good to know. So if I understand correctly, you don't actually want the tasks to appear in calendar, and the bug is that Calendar _is_ looking for them?

Could you export one of the tasks (make sure it doesn't have any personal info) and paste the file contents here? You can do that using Evolution. I assume that they will be `VTODO` type, which Calendar doesn't support. If so, then the question is then why Calendar is looking at that data source in the first place.

Is the account that you added a task list or a calendar? And did you add it from the Calendar app, the Tasks app, or Online Accounts settings?
